tests/Metrics/MetricsTest.cpp

```diff
@@ -257,6 +263,65 @@ template<typename T> void gauge_test() {
   T zdo = .1, zero = 0., one = 1.;
   Gauge g(zero, "gauge_1", "Gauge 1");
 
+  using namespace std;
+  using namespace std::chrono;
+
+  random_device rnd_device;
+  mt19937 mersenne_engine {rnd_device()};  // Generates random integers
+  uniform_real_distribution<T> dist {T(1), T(100)};
+  vector<T> vr(1000);
+  auto gen = [&dist, &mersenne_engine](){
+               return dist(mersenne_engine);
+             };
+  generate(vr.begin(), vr.end(), gen);
+
+  auto start = high_resolution_clock::now();
+  size_t const p = 10;
+  size_t const part = vr.size()/p;
+  std::vector<std::future<void>> f;
+
+  g = one;
+  for (size_t i = 0; i < p; ++i) {
+    f.push_back(
+      async(launch::async,
+            [&] {
+              for (size_t j = 0; j < part; ++j) {
+                g += vr.at(j);
+                g -= vr.at(j);
+              }}));
+  }
+  for (auto& i : f) {
+    i.wait();
+  }
+  std::cout << duration<double>(high_resolution_clock::now()-start).count() << std::endl;
+  
+  if constexpr (std::is_same<T,float>::value) {
+    ASSERT_TRUE(1.f - g.load() < 1.e-6f);
+  } else {
+    ASSERT_TRUE(1. - g.load() < 1.e-15);
+  }
+
+  g = one;
+  for (size_t i = 0; i < p; ++i) {
+    f.push_back(
+      async(launch::async,
+            [&] {
+              for (size_t j = 0; j < part; ++j) {
+                g *= vr.at(j);
+                g /= vr.at(j);
+              }}));
+  }
+  for (auto& i : f) {
+    i.wait();
+  }
+
+  if constexpr (std::is_same<T,float>::value) {
+    ASSERT_TRUE(1.f - g.load() < 1.e-6f);
```

Comment:
gtests have a sane floating point equality built in:
```suggestion
    ASSERT_FLOAT_EQ(1.f, g.load());
```

That will also fix the missing `abs()` call. ;)

Reply:
No, not reliable enough on windows